### PR TITLE
vss: remove dependency on live system during backup

### PR DESCRIPTION
**Pull request by sebsura**
### Thank you for contributing to the Bareos Project!

Sometimes bareos messes up and does not use the correct vss paths to backup data.  This PR identifies those situations and fixes them. 

This PR also:
* better supports backups with volume mount points,
* removes old comments from the windows installer,
* enables autocomplete for VolStatus = disabled in the update command,
* changed how st_ctime is calculated on windows,
* removes support for pre-vista vss clients (bareos did not support those systems for some time now), and
* made some strides in refactoring the compat layer

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: core/src/tests/compat_old_conversion.cc [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+  Dmsg1(debuglevel, "Leave make_wchar_win32_path=%s\n", pwszBuf);
+  return (POOLMEM*)pwszBuf;
+}
+
+int make_win32_path_UTF8_2_wchar(POOLMEM*& pszUCS,
+                                 const char* pszUTF,
+                                 BOOL* pBIsRawPath /*= NULL*/)
+{
+  int nRet;
+  // FUTURE NOTE: no caching inside the tests!
```
Comment (pstorz) [inline core/src/tests/compat_old_conversion.cc:342]: Is the following block still required?

File: core/src/tests/compat_old_conversion.cc [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+  // FUTURE NOTE: always use 32K conversions
+#define USE_WIN32_32KPATHCONVERSION
+#ifdef USE_WIN32_32KPATHCONVERSION
+  // Add \\?\ to support 32K long filepaths
+  pszUCS = make_wchar_win32_path(pszUCS, pBIsRawPath);
+#else
+  if (pBIsRawPath) { *pBIsRawPath = FALSE; }
+#endif
+
+  // FUTURE NOTE: no caching inside the tests!
```
Comment (pstorz) [inline core/src/tests/compat_old_conversion.cc:377]: Same here; Is this bock needed?

File: core/src/tests/compat_old_conversion.cc (hunk 1 above)
Comment (sebsura) [reply core/src/tests/compat_old_conversion.cc:342]: No.  Ill remove them.  This is just a note that I changed the "old" code slightly to make it suitable for regression testing.  But we do not need the actual code for that; just the comment is enough.

**Review by pstorz — APPROVED** [review]
Good work!

**Comment by rickmoua** [comment]
We are getting this error, it doesn't do to all windows systems but some systems we get this error.  JobId 18896: Fatal error: VSS was not initialized properly.

**Pull request merged**